expression/builtin_string.go

```diff
@@ -1366,3 +1367,126 @@ func builtinFindInSet(args []types.Datum, _ context.Context) (d types.Datum, err
 	}
 	return
 }
+
+type fieldFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *fieldFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	return &builtinFieldSig{newBaseBuiltinFunc(args, ctx)}, errors.Trace(c.verifyArgs(args))
+}
+
+type builtinFieldSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinFieldSig) eval(row []types.Datum) (types.Datum, error) {
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	return builtinField(args, b.ctx)
+}
+
+// See http://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_field
+func builtinField(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	// args[0] -> str
+	// args[1:] -> str list
```

Comment:
Not exactly, args[0..n] need not be a string. 
I think the comment should be:

> Returns the index (position) of str in the str1, str2, str3, ... list. Returns 0 if str is not found.
> If all arguments to FIELD() are strings, all arguments are compared as strings. If all arguments are numbers, they are compared as numbers. Otherwise, the arguments are compared as double.